File: TelerikUIExamplesInSwift/TelerikUIExamplesInSwift/SelectedPointLabel.swift
//
//  SelectedPointLabel.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import UIKit

class SelectedPointLabel: CALayer {
    
    var labelStyle: TKChartPointLabelStyle?
    var text: NSString?

    override func drawInContext(ctx: CGContext!) {
        UIGraphicsPushContext(ctx)
        let fill = self.labelStyle!.fill
        let stroke = TKStroke(color: UIColor.blackColor())
        let shape = TKBalloonShape(arrowPosition: TKBalloonShapeArrowPosition.Bottom, size: CGSizeMake(self.bounds.size.width - stroke.width, self.bounds.size.height - stroke.width))
        shape.drawInContext(ctx, withCenter: CGPointMake(CGRectGetMidX(self.bounds), CGRectGetMidY(self.bounds)), drawings: [fill, stroke])
        let textRect = CGRectMake(self.bounds.origin.x, self.bounds.origin.y - self.labelStyle!.insets.top, self.bounds.size.width, self.bounds.size.height + self.labelStyle!.insets.bottom)
        var paragraphStyle = NSMutableParagraphStyle.defaultParagraphStyle().mutableCopy() as NSMutableParagraphStyle
        paragraphStyle.alignment = self.labelStyle!.textAlignment
        let attributes: NSDictionary = [NSFontAttributeName: UIFont.systemFontOfSize(16),
            NSForegroundColorAttributeName: self.labelStyle!.textColor,
            NSParagraphStyleAttributeName: paragraphStyle]
        
        self.text!.drawWithRect(textRect, options: NSStringDrawingOptions.UsesLineFragmentOrigin, attributes: attributes, context: nil)
        UIGraphicsPopContext()
    }
    
}

File: Examples/Examples/Chart/PointLabels/CustomLabelRender/SelectedPointLabel.cs
using System;
using System.Drawing;

using Foundation;
using UIKit;
using CoreAnimation;
using CoreGraphics;

using TelerikUI;

namespace Examples
{
	public class SelectedPointLabel : CALayer
	{
		public TKChartPointLabelStyle LabelStyle {
			get;
			set;
		}

		public string Text {
			get;
			set;
		}

		public bool IsOutsideBounds {
			get;
			set;
		}

		public SelectedPointLabel()
		{
			this.NeedsDisplayOnBoundsChange = true;
			this.ContentsScale = UIScreen.MainScreen.Scale;
			this.DrawsAsynchronously = true;
		}

		public override void DrawInContext (CGContext ctx)
		{
			UIGraphics.PushContext (ctx);
			CGRect bounds = this.Bounds;
			TKFill fill = this.LabelStyle.Fill;
			TKStroke stroke = new TKStroke (UIColor.Black);
			TKBalloonShape shape = new TKBalloonShape (TKBalloonShapeArrowPosition.Bottom ,new CGSize(bounds.Size.Width - stroke.Width, bounds.Size.Height - stroke.Width));
			CGRect textRect;
			if (this.IsOutsideBounds == true) {
				shape.ArrowPosition = TKBalloonShapeArrowPosition.Top;
				textRect = new CGRect (bounds.Left, bounds.Top - this.LabelStyle.Insets.Top + shape.ArrowSize.Height, bounds.Size.Width, bounds.Size.Height + this.LabelStyle.Insets.Bottom);
			} else {
				textRect = new CGRect (bounds.Left, bounds.Top - this.LabelStyle.Insets.Top, bounds.Size.Width, bounds.Size.Height + this.LabelStyle.Insets.Bottom);
			}

			shape.DrawInContext (ctx, new CGPoint (bounds.GetMidX (), bounds.GetMidY ()), new TKDrawing[]{ fill, stroke });
			NSMutableParagraphStyle paragraphStyle = new NSMutableParagraphStyle ();
			paragraphStyle.Alignment = this.LabelStyle.TextAlignment;
			NSDictionary attributes = new NSDictionary (UIStringAttributeKey.Font, UIFont.SystemFontOfSize (18),
				UIStringAttributeKey.ForegroundColor, this.LabelStyle.TextColor, 
				UIStringAttributeKey.ParagraphStyle, paragraphStyle);

			NSString text = new NSString (this.Text);
			text.WeakDrawString (textRect, NSStringDrawingOptions.TruncatesLastVisibleLine | NSStringDrawingOptions.UsesLineFragmentOrigin, attributes, null);
			UIGraphics.PopContext ();
		}
	}
}

